zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -720,6 +726,8 @@ object Http {
     def apply[R, E, B](f: A => HExit[R, E, B]): Http[R, E, A, B] = FromFunctionHExit(f)
   }
 
+  private final case class When[R, E, A, B](f: A => Boolean, other: Http[R, E, A, B]) extends Http[R, E, A, B]
```

Comment:
We can define `IfThenElse` instead and can derive when from it. Let's discuss it.
cc: @tusharmath

Reply:
We can use `Http.fromFunction[A]` instead I think.